### DISCUSSION TITLE
Stat card value to display a sum

### DISCUSSION THREAD

**Discussion by jleech1992** [Budibase/budibase#13617] [Help] [answered]
Hello there,

Bare with me as I'm learning the ropes but I am currently trying Budibase and unable to achieve something that's probably simple. 

I have created a stat card labelled 'Total Sales' and in the value section, I'm trying to get it to display the sum of how many units there are in the 'RS Units' column in the table I have created in the data section of Budibase. (I have fake data in the table)
I am using ChatGPT as a guidance and so far it has provided unsuccessful coding whether it is Bindings or JS coding, the results I keep getting is either '0' or 'Error whilst executing JS'.

It would be greatly appreciated if an expert could help me to achieve this.  

Table name in data section: Business Finance Data
Row name: RS Units

Thanks in advance.

**Comment by mikesealey** [comment, marked as answer]
Hi @jleech1992 

I've been looking into this for you. Don't worry about not being able to achieve something simple, it's only simple when you know the answer! Also a quick note on rows vs columns - rows go horizontally across the table (think rows of seats in a theatre), and columns go vertically [(Like the columns of the Parthenon)](https://en.wikipedia.org/wiki/Parthenon) .

The way I've tackled this is by using a `Data Provider` component, and putting the `Stat Card` inside that. Set the `Data Provider` to fetch from your table using the `Data` option.

_Component Tree_
![image](https://github.com/Budibase/budibase/assets/39269380/7b44f92b-fd11-4ffd-819d-8c7611d531f1)

_Data Provider Settings_
![image](https://github.com/Budibase/budibase/assets/39269380/8ff9a5ff-633d-4488-8cd3-05ae956423bb)



Then, in the `Value` field of your `Stat Card` switch to JavaScript. 
`let rows = $("New Data Provider.Rows")`
_This gets all of the rows in the table you set in the data provider, as an array of objects. You can see what it looks like by clicking on your Data Provider in the bindings menu and hovering over "Rows"_
![image](https://github.com/Budibase/budibase/assets/39269380/15a0e81b-b5a0-4e85-805d-66f4e97d40fd)

`let total = 0`
_This variable will be used to tally up everything in your RS_Units column_

```JS
rows.forEach(row => {
  total += row.RS_Units
})
```
_This `forEach()` function loops through the array of objects and adds the RS_Units number to the current value of `total`_

And lastly, make sure you are returning the value you want to display with
`return total`

The overall JavaScript function should look similar to this, but be mindful that your variables, table names, and components may have different names.

I also changed the column name "RS Units" to "RS_Units" in the table, so that prop-drilling would be easier (see Line 5)
![image](https://github.com/Budibase/budibase/assets/39269380/5cb544c4-57c5-49be-bde9-bd40c17ac977)

I hope this helps get you on your way. In the meantime be sure to have [the Budibase documentation](https://docs.budibase.com/docs/what-is-budibase) and [JavaScript documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript) nearby, and if you haven't already, come and join us on [the discord server](https://discord.com/invite/ZepTmGbtfF).
